gamehub-frontend/src/pages/GameGallery/GameGallery.tsx

```diff
@@ -1,16 +1,25 @@
-
-import {Game, User} from "../../types.ts";
+import { Box, FormControl, InputLabel, MenuItem, Select, TextField, Typography } from '@mui/material';
 import GameCard from "../../components/GameCard/GameCard.tsx";
-import {Box} from '@mui/material';
+import { Game, User } from "../../types.ts";
+import { useState } from "react";
 
 type GameGalleryProps = {
-    games: Game[]
-    user: User
-    addGameToLibrary: (gameId: string) => void
-    deleteGameFromLibrary: (gameId: string) => void
-}
+    games: Game[];
+    user: User;
+    addGameToLibrary: (gameId: string) => void;
+    deleteGameFromLibrary: (gameId: string) => void;
+};
 
 export default function GameGallery(props: Readonly<GameGalleryProps>) {
+    const [searchQuery, setSearchQuery] = useState('');
+    const [selectedPlatform, setSelectedPlatform] = useState<string>('');
+
+    const platforms = Array.from(new Set(props.games.flatMap(game => game.platforms)));
+
+    const filteredGames = props.games.filter(game =>
+        game.title.toLowerCase().includes(searchQuery.toLowerCase()) &&
```

Comment:
a small tipp:
If there's any chance that game.title or searchQuery could be null or undefined, please use '?'
`game.title?.toLowerCase().includes(searchQuery?.toLowerCase() || '') && `

If it is certain that 'game.title' or 'searchQuery' has always non-null strings, don't need to use '?'

Reply:
thanks  :)

Reply:
You're welcome :)